added Grown visualizer to the Generator documentation

NiLSPACE opened the pull request.

madmaxoft commented on docs/Generator.html:
```diff
@@ -429,9 +431,41 @@
+<table>
+	<tr>
+		<td rowspan="5">
+			<canvas id="stage" width="400" height="400"></canvas>
```
Make it larger, at least 600 * 600. And limit the zoom so that one cannot zoom in when the array size is larger than the canvas size.

NiLSPACE replied:
How about I give a warning that rendering larger than 600x600 can lag the browser?

madmaxoft replied:
That's pretty useless, reading the warning and clicking OK will require more effort than just waiting for the render. No, a simple disabling the button would be much better, in my opinion.

madmaxoft commented on docs/js/ValueMap.js:
```diff
@@ -0,0 +1,139 @@
+		context.fillText("Size: " + (this.sizeX - 1) + "x" + (this.sizeZ - 1), 5, 10);
+		context.restore();
+	}
+}
```
missing newline

madmaxoft commented on docs/js/grown.js:
```diff
@@ -0,0 +1,70 @@
+}
+
+
+window.onload = init;
```
Missing newline

NiLSPACE commented:
I've added the missing newlines, made the canvas bigger and added a warning if the map gets too big. Also, I've added some some more information about the visualizer that rendering the map takes way longer than the actual zooming/smoothing.

NiLSPACE commented:
Done.

madmaxoft commented on docs/Generator.html:
```diff
@@ -429,9 +431,41 @@
 
+<br />
+
+Here is a visualizer where you can play with the algorithm that the Grown biome generator uses. I'd like to point out that rendering the map takes way longer than the zoom/smooth operation.
```
I don't think using "I'd like" is good in this text - no other part of the text uses that form. How about "Note that ..."?

NiLSPACE replied:
Yeah, that sounds much better.

madmaxoft commented on docs/js/grown.js:
```diff
@@ -0,0 +1,79 @@
+
+
+function btnAutomatic(target) {
+	target.disabled = true;
```
Does this disable the Auto button? If so, I don't see any code to re-enable it.

NiLSPACE replied:
It does at line 65 :)

madmaxoft replied:
Yea, as I said, I'm blind in JS

madmaxoft commented:
/me likes, no objections from me towards a merge.

NiLSPACE commented:
Great, then I'll merge this :)

SphinxC0re commented:
I recognized, your are using es6 syntax. As long as you aren't using any transpiler this is not compatible with all systems.

NiLSPACE commented:
It should be compatible with all modern browsers. I've tried it on Windows with Firefox, Edge, Chrome. Linux on Firefox and Windows Phone with Edge.

SphinxC0re commented:
https://babeljs.io/
